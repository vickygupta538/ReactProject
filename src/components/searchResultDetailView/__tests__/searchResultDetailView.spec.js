import React from "react";
import renderer from "react-test-renderer";
import { testStore } from "../../../../test/utils";
import SearchResultDetailView from "../index";
import { Provider } from "react-redux";

const initialState = {
  characters: [
    {
      name: "Luke Skywalker",
      height: "172",
      mass: "77",
      hair_color: "blond",
      skin_color: "fair",
      eye_color: "blue",
      birth_year: "19BBY",
      gender: "male"
    }
  ],
  selectedCharacterIndex: 0
};

const setUp = () => {
  const store = testStore(initialState);
  const component = renderer.create(
    <Provider store={store}>
      <SearchResultDetailView />
    </Provider>
  );

  return component;
};

describe("SearchResultDetailView Component Test Cases", () => {
  let component;

  beforeEach(() => {
    component = setUp(initialState);
  });

  it("should match snapshot renderer", () => {
    const wrapper = component.toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render without errors", () => {
    const wrapper = component.root.find(
      el => el.type === "section" && el.props["data-test"] === "SearchResultDetailViewComponent"
    );
    expect(wrapper).toBeTruthy();
  });
});
