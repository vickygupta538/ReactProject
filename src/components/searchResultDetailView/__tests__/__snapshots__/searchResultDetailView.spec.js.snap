// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultDetailView Component Test Cases should match snapshot renderer 1`] = `
<section
  className="searchResultDetailView"
>
  <section
    data-test="SearchResultDetailViewComponent"
  >
    <h2>
      Luke Skywalker
    </h2>
    <div
      className="attributeHeading"
    >
      Hair Color
    </div>
    <div
      className="attributeContent"
    >
      blond
    </div>
    <div
      className="attributeHeading"
    >
      Skin Color
    </div>
    <div
      className="attributeContent"
    >
      fair
    </div>
    <div
      className="attributeHeading"
    >
      Eye Color
    </div>
    <div
      className="attributeContent"
    >
      blue
    </div>
    <div
      className="attributeHeading"
    >
      Birth Year
    </div>
    <div
      className="attributeContent"
    >
      19BBY
    </div>
    <div
      className="attributeHeading"
    >
      Gender
    </div>
    <div
      className="attributeContent"
    >
      male
    </div>
  </section>
</section>
`;
